### PR TITLE
Added php7 to travis build

### PR DESCRIPTION
**Pull request by fonsecas72**

### CODE REVIEW

**Comment by keradus** [comment]
I was looking for it quite ago but it wasn't ready for Travis. Glad to see it! Thanks!
I will merge it, maybe we allow to fail it in the future if incompatibilites will be too big.

Wow, so fast...

**Comment by fonsecas72** [comment]
Yeah, I though that we might wanted to known when it fails but ok, we sure don't want a red build when PHP7 is failing.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,11 @@ php:
     - 5.4
     - 5.5
     - 5.6
+    - 7.0
     - hhvm
+matrix:
+    allow_failures:
+        - php: 7.0
```
Comment (GrahamCampbell) [inline .travis.yml]: You should enable "fast finish" here.

Comment (fonsecas72) [inline .travis.yml]: Yeah that might be better, thanks.

**Comment by keradus** [comment]
Thank you @fonsecas72.

**Comment by stof** [comment]
@keradus is it time to open a bunch of tickets about the new PHP 7 syntaxes which need to be added in the testsuite (and then be handled by the code) ? There are a bunch of things not handled properly by the current codebase which won't be catched by the testsuite because it does not use the new features

**Comment by keradus** [comment]
TBH I'm not sure about it. The new PHP7 syntax may change until first stable release.

**Comment by GrahamCampbell** [comment]
Yeh, I think we should at least wait until we've got "beta" tags from php. Normally we get "alpha" tags first, then "beta" tags, then "RC" releases from php, don't we?

**Comment by keradus** [comment]
And didn't PHP7 plan to replace native stream tokenizer with AST ?

**Comment by GrahamCampbell** [comment]
It's not replaced. AST is just an abstraction applied after tokenization.

**Comment by GrahamCampbell** [comment]
> plan to replace

And the AST was implemented ages ago. If I remember, it was essentially done when the ng branch got merged to the master.

**Comment by ceeram** [comment]
AST is nice, but not very usable for php-cs-fixer imo. AST does not care
about whitespace and comments, so in php7 would still need to use the
current tokens

2015-03-02 23:40 GMT+01:00 Graham Campbell <notifications@github.com>:

**Comment by GrahamCampbell** [comment]
@ceeram, yeh, exactly. AST is useful for working out what the code actually means rather than if it's layed out correctly, though, it could potentially be useful for our transformers. I haven't really looked at php's ast. Does it have an interface for the end user like the tokenization does?

**Comment by keradus** [comment]
I remember ages ago that nikic told he would handle non-meaningful parts of code too, still waiting for it.
Good to know that nikic's AST woudn't replace stream tokenizer, just be an addition to it.

**Comment by GrahamCampbell** [comment]
Yeh, the AST requires the tokenization step.

**Comment by stof** [comment]
@GrahamCampbell accessing the AST from PHP code is not possible with the PHP core alone. However, Nikic wrote an extension exposing it to userland: https://github.com/nikic/php-ast

However, I don't think the AST would help fixing CS. The AST represents the meaning of the code, not the way it is written. whitespaces are not relevant for the AST at all for instance (especially given that I don't know whether the exposed AST will correspond to the initial AST created from the tokenized code or to the processed AST after applying some optimizations).

**Comment by keradus** [comment]
Ref: https://github.com/nikic/PHP-Parser/issues/41

**Comment by stof** [comment]
@keradus this is not about the PHP 7 AST though

**Comment by keradus** [comment]
It's about whitespaces etc in nikic's AST

**Pull request closed without merging**